### PR TITLE
Add Perl sample code to get Asset by a filter and invoke Asset Cleanu…

### PR DESCRIPTION
**Pull request by pedrorcs**
Add Perl sample code to get Asset by a filter and invoke Asset Cleanup API to delete those assets returned by the filter.

### CODE REVIEW

File: snippets/perl/README.md [hunk 1]
```diff
@@ -24,3 +24,13 @@ Catalog Image Details:
 
 - Use the following command to obtain the catalog image details from your NetBackup Master server:
   - `perl get_nb_images.pl -nbmaster <master_server> -username <username> -password <password> [-domainname <domain_name>] [-domaintype <domain_type>] [--verbose]`
+
+
+Asset Cleanup Detais:
```
Comment (kk-github-personal) [inline snippets/perl/README.md]: Typo: Detais -> Details.

File: snippets/perl/README.md [hunk 2]
```diff
@@ -24,3 +24,13 @@ Catalog Image Details:
...
+- Use the following command to query Assets by a user given specified filter, and cleanup time to delete all Assets returned by the filter. The cleanupTime field is an 
+ISO 8601 formatted UTC timestamp.
+- Keep in mind that those Assets returned by the filter will be only delete if, The last discovered time of the Asset is older than the given cleanupTime and there is 
+not a subscription associated with this asset.
+
+  - Example: perl post_nb_cleanup_asset.pl -nbmaster <master_server> -username <username> -password <pass> --filter "workloadType eq 'VMware'" --cleanuptime 2018-06-29T15:58:45.678Z
```
Comment (kk-github-personal) [inline snippets/perl/README.md]: I think we are mixing options with '-' and '--'. If '-' is followed elsewhere, we should probably stick to that for consistency.

File: snippets/perl/netbackup.pm [hunk 3]
```diff
@@ -316,5 +317,183 @@ sub displayCatalogImages {
...
+
+    $data = decode_json($response->content);
+    my $pretty = JSON->new->pretty->encode($data);
+    return $pretty;
+  }
+  else {
+    print "ERROR :: Get Assets Request Failed!\n";
+    print "HTTP GET error code: ", $response->code, "\n";
+    print "HTTP GET error message: ", $response->message, "\n";
+    #die;  let this fall through to cleanup code
```
Comment (kk-github-personal) [inline snippets/perl/netbackup.pm]: Commented code can be removed.

File: snippets/perl/post_nb_cleanup_asset.pl [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+use lib '.';
+
+use netbackup;
+use strict;
+use warnings;
+use Getopt::Long qw(GetOptions);
+use JSON;
+
+sub printUsage {
+  print "\nUsage : perl get_nb_jobs.pl -nbmaster <master_server> -username <username> -password <password> --filter <filter> --cleanuptime <cleanuptime> [-domainname <domain_name>] [-domaintype <domain_type>] [--verbose]\n\n";
```
Comment (kk-github-personal) [inline snippets/perl/post_nb_cleanup_asset.pl]: Use points to get_nb_jobs.

File: snippets/perl/post_nb_cleanup_asset.pl [hunk 5]
```diff
@@ -0,0 +1,75 @@
+#Load module netbackup.pm from current directory
```
Comment (kk-github-personal) [inline snippets/perl/post_nb_cleanup_asset.pl]: Should the name be post_nb_assets_cleanup?

File: snippets/perl/post_nb_cleanup_asset.pl (hunk 4 above)
Comment (kk-github-personal) [reply snippets/perl/post_nb_cleanup_asset.pl]: Also keep the username password domainname and domaintype together.

File: snippets/perl/post_nb_cleanup_asset.pl [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+my $jsonstring = netbackup::getAssetsByFilter($fqdn_hostname, $myToken, $filter);
+my $page_count = decode_json($jsonstring)->{'meta'}->{'pagination'}->{'count'};
+
+if ($page_count ne 0 ){
+   print "\nNetBackup Assets returned by filter: $filter\n";
+   netbackup::displayAssets($jsonstring);
+   netbackup::cleanAssets($fqdn_hostname, $myToken, $jsonstring, $cleanuptime);
+} else {
+   print "\n Your filter: $filter did not reutnr any asset."
```
Comment (kk-github-personal) [inline snippets/perl/post_nb_cleanup_asset.pl]: reutnr -> return

File: snippets/perl/README.md (hunk 2 above)
Comment (pedrorcs) [reply snippets/perl/README.md]: I kept the single dash "-"

File: snippets/perl/README.md (hunk 1 above)
Comment (pedrorcs) [reply snippets/perl/README.md]: Fixed.

File: snippets/perl/netbackup.pm (hunk 3 above)
Comment (pedrorcs) [reply snippets/perl/netbackup.pm]: Removed.

File: snippets/perl/post_nb_cleanup_asset.pl (hunk 5 above)
Comment (pedrorcs) [reply snippets/perl/post_nb_cleanup_asset.pl]: I will rename the file to post_nb_asset_cleanup.pl

File: snippets/perl/post_nb_cleanup_asset.pl (hunk 4 above)
Comment (pedrorcs) [reply snippets/perl/post_nb_cleanup_asset.pl]: domainname and domaintype, those are optional fields.

Comment (pedrorcs) [reply snippets/perl/post_nb_cleanup_asset.pl]: Fixed the name of the script.

File: snippets/perl/post_nb_cleanup_asset.pl (hunk 6 above)
Comment (pedrorcs) [reply snippets/perl/post_nb_cleanup_asset.pl]: Fixed.

**Comment by pedrorcs** [comment]
Applied all fixes suggested.

**Review by kk-github-personal — DISMISSED** [review]

File: snippets/perl/README.md [hunk 7]
```diff
@@ -24,3 +24,13 @@ Catalog Image Details:
 
 - Use the following command to obtain the catalog image details from your NetBackup Master server:
   - `perl get_nb_images.pl -nbmaster <master_server> -username <username> -password <password> [-domainname <domain_name>] [-domaintype <domain_type>] [--verbose]`
+
+
+Asset Cleanup Details:
+- Use the following command to query Assets by a user given specified filter, and cleanup time to delete all Assets returned by the filter. The cleanupTime field is an 
+ISO 8601 formatted UTC timestamp.
+- Keep in mind that those Assets returned by the filter will be only delete if, The last discovered time of the Asset is older than the given cleanupTime and there is
```
Comment (jaya-korlepara) [inline snippets/perl/README.md]: Couple of typos here - "will be deleted only if, the last discovered"

File: snippets/perl/README.md [hunk 8]
```diff
@@ -24,3 +24,13 @@ Catalog Image Details:
 
 - Use the following command to obtain the catalog image details from your NetBackup Master server:
   - `perl get_nb_images.pl -nbmaster <master_server> -username <username> -password <password> [-domainname <domain_name>] [-domaintype <domain_type>] [--verbose]`
+
+
+Asset Cleanup Details:
+- Use the following command to query Assets by a user given specified filter, and cleanup time to delete all Assets returned by the filter. The cleanupTime field is an 
+ISO 8601 formatted UTC timestamp.
+- Keep in mind that those Assets returned by the filter will be only delete if, The last discovered time of the Asset is older than the given cleanupTime and there is 
+not a subscription associated with this asset.
```
Comment (jaya-korlepara) [inline snippets/perl/README.md]: "there is no subscription associated"

File: snippets/perl/netbackup.pm [hunk 9]
```diff
@@ -316,5 +317,182 @@ sub displayCatalogImages {
...
+  else {
+    print "ERROR :: Get Assets Request Failed!\n";
+    print "HTTP GET error code: ", $response->code, "\n";
+    print "HTTP GET error message: ", $response->message, "\n";
+  }
+
+}
+
+#
+## This function displays the Json content returned from Asset API
```
Comment (jaya-korlepara) [inline snippets/perl/netbackup.pm]: "JSON"

File: snippets/perl/netbackup.pm [hunk 10]
```diff
@@ -316,5 +317,182 @@ sub displayCatalogImages {
...
+## using query filter in a tabular format
+##
+sub displayAssets {
+
+  my $arguments_count = scalar(@_);
+  if ($arguments_count != 1) {
+     print "ERROR :: Incorrect number of arguments passed to displayAssets()\n";
+     print "Usage : displayAssets( <Json content returned from Assets API> ) \n";
```
Comment (jaya-korlepara) [inline snippets/perl/netbackup.pm]: "JSON"

**Review by jaya-korlepara — DISMISSED** [review]
JSON is misspelled at few places.

File: snippets/perl/README.md (hunk 7 above)
Comment (pedrorcs) [reply snippets/perl/README.md]: Fixed

File: snippets/perl/README.md (hunk 8 above)
Comment (pedrorcs) [reply snippets/perl/README.md]: Fixed

File: snippets/perl/netbackup.pm (hunk 10 above)
Comment (pedrorcs) [reply snippets/perl/netbackup.pm]: Fixed

**Comment by pedrorcs** [comment]
Applied the fixes suggested by @jaya-korlepara and fixed merging conflicts in the README.pm file.

**Review by jaya-korlepara — APPROVED** [review]

**Pull request merged**